dask_cuda/cuda_worker.py

```diff
@@ -221,53 +221,102 @@ def del_pid_file():
-                        enable_rdmacm=enable_rdmacm,
-                        net_devices=net_devices,
-                        cuda_device_index=i,
-                    )
-                },
-                data=data(nvml_device_index(i, cuda_visible_devices(i))),
-                worker_class=worker_class,
-                **kwargs,
-            )
-            for i in range(nprocs)
-        ]
+        if nanny:
+            t = Nanny
+            kwargs = {"worker_port": None, "listen_address": None, **kwargs}
+            self.nannies = [
+                Nanny(
+                    scheduler,
+                    scheduler_file=scheduler_file,
+                    nthreads=nthreads,
+                    dashboard=dashboard,
+                    dashboard_address=dashboard_address,
+                    http_prefix=dashboard_prefix,
+                    loop=loop,
+                    resources=resources,
+                    memory_limit=memory_limit,
+                    interface=_get_interface(interface, host, i, net_devices),
+                    host=host,
+                    preload=(list(preload) or []) + ["dask_cuda.initialize"],
+                    preload_argv=(list(preload_argv) or []) + ["--create-cuda-context"],
+                    security=security,
+                    env={"CUDA_VISIBLE_DEVICES": cuda_visible_devices(i)},
+                    plugins={
+                        CPUAffinity(
+                            get_cpu_affinity(
+                                nvml_device_index(i, cuda_visible_devices(i))
+                            )
+                        ),
+                        RMMSetup(
+                            rmm_pool_size,
+                            rmm_maximum_pool_size,
+                            rmm_managed_memory,
+                            rmm_async,
+                            rmm_log_directory,
+                        ),
+                    },
+                    name=name
+                    if nprocs == 1 or name is None
+                    else str(name) + "-" + str(i),
+                    local_directory=local_directory,
+                    config={
+                        "distributed.comm.ucx": get_ucx_config(
+                            enable_tcp_over_ucx=enable_tcp_over_ucx,
+                            enable_infiniband=enable_infiniband,
+                            enable_nvlink=enable_nvlink,
+                            enable_rdmacm=enable_rdmacm,
+                            net_devices=net_devices,
+                            cuda_device_index=i,
+                        )
+                    },
+                    data=data(nvml_device_index(i, cuda_visible_devices(i))),
+                    worker_class=worker_class,
+                    **kwargs,
+                )
+                for i in range(nprocs)
+            ]
+        else:
+            t = Worker
+
+            self.nannies = [
+                t(
+                    scheduler,
```

Comment:
Are we able to reuse this for the `if` block above as well? If so, can we move this out of the branch and call once?

Reply:
+1 to John's comment. I think what you need is:

```python
t = Nanny if nanny else Worker

self.nannies = [...]
```

Reply:
Actually here we probably want `CUDAWorker`, rather than `Worker`.